src/cpp/Memory.h

```diff
@@ -83,25 +82,25 @@ namespace libstdhl
 
         //
         // shared object creation utility which allocates only new objects
-        // if they are not already cached in the 'make_cache()' facility
-        // through 'make_hash()'
+        // if they are not already cached in the 'cache()' facility
+        // through 'hash()'
         //
         template < typename T, typename... Args >
         inline std::shared_ptr< T > get( Args&&... args )
         {
             T tmp = T( std::forward< Args >( args )... );
 
-            auto cache = tmp.make_cache().find( tmp.make_hash() );
-            if( cache != tmp.make_cache().end() )
+            const auto tmp_hash = tmp.hash();
+
+            auto cache = tmp.cache().find( tmp_hash );
```

Comment:
const + rename `cache` to `it`, because find returns an iterator.

Reply:
ah you are right ... already fixed